ambari-server/src/main/python/ambari_server/setupSso.py

```diff
@@ -58,7 +60,7 @@ def validateOptions(options):
       errors.append("Missing option: --sso-provider-url")
     if not options.sso_public_cert_file:
       errors.append("Missing option: --sso-public-cert-file")
-    if options.sso_provider_url and not re.search(REGEX_HOSTNAME_PORT, options.sso_provider_url):
+    if options.sso_provider_url and not re.search(REGEX_URL, options.sso_provider_url):
```

Comment:
why not re.match? 
https://docs.python.org/3/library/re.html#search-vs-match
any string that _contains_ url would pass the condition